src/components/candevice/candevice.cpp

```diff
@@ -34,6 +44,19 @@ bool CanDevice::init(const QString& backend, const QString& interface)
     return d->_initialized;
 }
 
+bool CanDevice::init()
+{
+    Q_D(CanDevice);
+    if (d->_props.count(d->_backendProperty) != 1 ||
+            d->_props.count(d->_backendProperty) != 1)
+    {
+        return false;
+    }
+
+    return init(d->_props[d->_backendProperty].toString(),
```

Comment:
I would just return once, i.e.:

```
return (...) ? false : init(...);
```

Reply:
The return statement will be a 4-liner with condition having 2-level nested parentheses... are you sure? ;)

Reply:
Restructure and assert.

```c++
assert(nullptr != d);

const auto& props   = d->_props;
const auto& backend = d->_backendProperty;
const auto& iface   = d->_interfaceProperty;

// put here what does this condition mean
const bool cond = (props.count(backend) != 1)
               || (props.count(iface) != 1);

return (true == cond)
          ? false
          : init(props[backend].toString()
               , props[iface].toString()
               , false);
```

Reply:
changed as requested